### PR TITLE
Added Pip as requirement for functions

### PR DESCRIPTION
**Pull request by alchemistake**
## Description
Changed how the import of pip works and then added an additional optional dependency to add support for pip-less pythons such as distroless.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change. 
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring. 
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by emelvarol-cognite — APPROVED** [review]
Lgtm 🚀😎

File: cognite/client/_api/functions.py [hunk 1]
```diff
@@ -721,6 +719,8 @@ def _validate_and_parse_requirements(requirements: List[str]) -> List[str]:
     parsed_reqs: List[str] = []
     for req in requirements:
         try:
+            from pip._internal.req.constructors import install_req_from_line
```
Comment (erlendvollset) [inline cognite/client/_api/functions.py:722]: Don't import inside the try/except, or the import error gets transformed into a ValueError. You're also reimporting the module  for every iteration in the loop.

We also have a local_import function in utils which will raise an exception with a more friendly error message. I'd prefer if you used that.

The `except Exception` is also bad practice. But that can be addressed in a different PR.

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ Changes are grouped as follows
 - `Fixed` for any bug fixes.
 - `Security` in case of vulnerabilities.
 
+## [4.0.2] - 2022-08-19
```
Comment (erlendvollset) [inline CHANGELOG.md:17]: Also specify that you've added a new optional dependency and what the extra is called (functions).

**Review by erlendvollset — CHANGES_REQUESTED** [review]

**Review by HMEiding — APPROVED** [review]
LGTM 🚀 Update the PR description?

**Review by erlendvollset — APPROVED** [review]
lgtm

**Comment by emelvarol-cognite** [comment]
@tuanng-cognite Can you review on behalf of geospatial? 🙏

**Review by tuanng-cognite — APPROVED** [review]

**Pull request merged**